vips/vip-171/vip-171-testnet.ts

```diff
@@ -0,0 +1,129 @@
+    reduceReservesBlockDelta: 840000,
+    acceptAdmin: false,
+  },
+  {
+    name: "vTUSD",
+    address: "0xEFAACF73CE2D38ED40991f29E72B12C74bd4cf23",
+    reduceReservesBlockDelta: 840000,
+    acceptAdmin: false,
+  },
+  {
+    name: "vUSDC",
+    address: "0xD5C4C2e2facBEB59D0216D0595d63FcDc6F9A1a7",
+    reduceReservesBlockDelta: 840000,
+    acceptAdmin: true,
+  },
+  {
+    name: "vUSDT",
+    address: "0xb7526572FFE56AB9D7489838Bf2E18e3323b441A",
+    reduceReservesBlockDelta: 840000,
+    acceptAdmin: false,
+  },
+  {
+    name: "vWBETH",
+    address: "0x35566ED3AF9E537Be487C98b1811cDf95ad0C32b",
+    reduceReservesBlockDelta: 840000,
+    acceptAdmin: false,
+  },
+];
+
+function filterAssets(assets: AssetConfig[]) {
+  return assets.filter(asset => asset.acceptAdmin);
+}
+
+const MARKETS_WITH_ACCEPT_ADMIN = filterAssets(CORE_MARKETS);
+
+export const vip171Testnet = () => {
+  const meta = {
+    version: "v2",
+    title: "VIP-171 VToken Upgrade of AIA",
+    description: `upgrade the implementation of the Vtoken core supportimg Automatic income allocation feature.`,
+    forDescription: "I agree that Venus Protocol should proceed with VToken Upgrade of AIA",
+    againstDescription: "I do not think that Venus Protocol should proceed with VToken Upgrade of AIA",
+    abstainDescription: "I am indifferent to whether Venus Protocol proceeds with VToken Upgrade of AIA or not",
+  };
+
+  return makeProposal(
+    [
+      ...MARKETS_WITH_ACCEPT_ADMIN.map(asset => {
+        return {
+          target: asset.address,
+          signature: "_acceptAdmin()",
+          params: [],
+        };
+      }),
+      ...CORE_MARKETS.map(asset => {
+        return {
+          target: asset.address,
+          signature: "_setImplementation(address,bool,bytes)",
+          params: [NEW_VBEP20_DELEGATE_IMPL, false, "0x"],
+        };
+      }),
+
+      ...CORE_MARKETS.map(asset => {
+        return {
+          target: asset.address,
+          signature: "setAccessControlManager(address)",
+          params: [ACCESS_CONTROL_MANAGER],
+        };
+      }),
+
+      {
+        target: ACCESS_CONTROL_MANAGER,
+        signature: "giveCallPermission(address,string,address)",
+        params: [ethers.constants.AddressZero, "setReduceReservesBlockDelta(uint256)", NORMAL_TIMELOCK],
+      },
+
+      {
+        target: ACCESS_CONTROL_MANAGER,
+        signature: "giveCallPermission(address,string,address)",
+        params: [ethers.constants.AddressZero, "setProtocolShareReserve(address)", NORMAL_TIMELOCK],
+      },
+
+      ...CORE_MARKETS.map(asset => {
+        return {
+          target: asset.address,
+          signature: "setReduceReservesBlockDelta(uint256)",
+          params: [asset.reduceReservesBlockDelta],
+        };
+      }),
+
+      ...CORE_MARKETS.map(asset => {
+        return {
+          target: asset.address,
+          signature: "setProtocolShareReserve(address)",
+          params: [TREASURY],
```

Comment:
This should be the new ProtocolShareReserve contract, right?

Reply:
This VIP will be executed after https://github.com/VenusProtocol/vips/pull/53, where the new ProtocolShareReserve contract will be configured. So, we should use that contract address here

Reply:
I think you mean #59 here?

Reply:
ACK
I'll update address after final deployments.